src/main/java/com/amazon/dlic/auth/http/saml/AuthTokenProcessorHandler.java

```diff
@@ -252,7 +242,7 @@ JWK createJwkFromSettings(Settings settings, Settings jwtSettings) throws Except
         String exchangeKey = settings.get("exchange_key");
 
         if (!Strings.isNullOrEmpty(exchangeKey)) {
```

Comment:
Before the JWT library change, the CXF library assumed that the exchangeKey was Base64 encoded:

1. https://github.com/opensearch-project/security/blob/2.11/src/main/java/com/amazon/dlic/auth/http/saml/AuthTokenProcessorHandler.java#L257
2. https://github.com/apache/cxf/blob/main/rt/rs/security/jose-parent/jose/src/main/java/org/apache/cxf/rs/security/jose/jws/JwsUtils.java#L109-L112
3. https://github.com/apache/cxf/blob/main/rt/rs/security/jose-parent/jose/src/main/java/org/apache/cxf/rs/security/jose/common/JoseUtils.java#L112-L114
4. https://github.com/apache/cxf/blob/main/rt/security/src/main/java/org/apache/cxf/rt/security/crypto/CryptoUtils.java#L669-L675

With the change to nimbus, the configured value should also be assumed to be base64 encoded. The nimbus call:

```
new OctetSequenceKey.Builder(exchangeKey.getBytes(StandardCharsets.UTF_8))
```

will perform Base64 encoded as part of the [constructor of the builder](https://github.com/felx/nimbus-jose-jwt/blob/master/src/main/java/com/nimbusds/jose/jwk/OctetSequenceKey.java#L179-L186) so the code above decodes the configured value, pads it (for backwards compatibility?) and then encodes again. 

The code to remove the Base64 decoding here would be needed for backwards compatibility so it cannot be removed.

Reply:
> The code to remove the Base64 decoding here would be needed for backwards compatibility so it cannot be removed.

I'm slightly confused. Are you suggesting we move forward with getUrlDecoder in place of getDecoder?

Reply:
Yes, `getUrlDecoder()` would be in line with the previous JWT functionality using cxf.

Reply:
I've also replaced with url decoder to keep it backwards compatible.